test-data/unit/check-classes.test

```diff
@@ -6677,3 +6677,20 @@ class B:
                                         # N: Perhaps you need "Callable[...]" or a callback protocol?
 
 [builtins fixtures/classmethod.pyi]
+
+[case testFinalClassWithAbstractAttributes]
+from abc import abstractmethod, ABCMeta
+from typing import final
+
+@final
+class A(metaclass=ABCMeta):  # E: Final class __main__.A has abstract attributes "bar", "foo"
```

Comment:
Please add also the same test without `ABCMeta`, and a test where abstract attribute is inherited (while the body of final class itself doesn't define abstract attributes).

Reply:
I've added 2 additional tests, let me know if they correspond to what you meant.